### ISSUE TITLE
[PRE REVIEW]: JASP for Audit: Bayesian Tools for the Auditing Practice

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2152]
**Submitting author:** @koenderks (<a href="http://orcid.org/0000-0002-5533-9349">Koen Derks</a>)
**Repository:** <a href="https://github.com/koenderks/jasp-desktop" target ="_blank">https://github.com/koenderks/jasp-desktop</a>
**Version:** v1.0.0
**Editor:** @drvinceknight
**Reviewers:** @trstewart212 , @fgeertman
**Managing EiC:** Daniel S. Katz

**Author instructions**

Thanks for submitting your paper to JOSS @koenderks. **Currently, there isn't an JOSS editor assigned** to your paper.

The author's suggestion for the handling editor is @arfon.

@koenderks if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission (please start at the bottom of the list).

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2152 with the following error: 

 Can't find any papers to compile :-(

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.84  T=6.25 s (336.9 files/s, 81078.4 lines/s)
-----------------------------------------------------------------------------------
Language                         files          blank        comment           code
-----------------------------------------------------------------------------------
SVG                                756            109             78         164025
R                                  285          17254           8563          66135
C++                                225          10938           3588          41762
JavaScript                          22           7823           4640          36457
QML                                242           5648           3451          31001
PO File                              7           9471             55          29357
Markdown                           211           4431             24          16905
C/C++ Header                       265           4321           4843          16081
JSON                                45            372              0          10753
CSS                                 10            389             94           3267
HTML                                 6             31              0           1638
ProGuard                             2             94             21            642
Bourne Shell                        11            138             64            458
Qt Project                           4             53              6            320
DOS Batch                            3            103             21            320
Python                               2             86             18            282
XML                                  3              0              2            110
PHP                                  1             38              5            110
WiX source                           1             28             15            105
C                                    1             10             28             41
YAML                                 1              0              2             14
make                                 1              2              0              4
Windows Resource File                1              0              0              1
-----------------------------------------------------------------------------------
SUM:                              2105          61339          25518         419788
-----------------------------------------------------------------------------------


Statistical information for the repository '2152' was gathered on 2020/03/09.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
AlexanderLyNL                    8           114             17            0.02
Anneliese Wild                   1             6              6            0.00
Bruno Boutin                     1           203            146            0.06
Don van den Bergh                4           583            401            0.18
Erik-Jan van Kestere            20          1911           1029            0.52
Frans Meerhoff                   1            89              1            0.02
FransMeerhoff                  145         19253          12428            5.64
FransTestAccount                 2           328            344            0.12
Jan G. Voelkel                   7            30              7            0.01
JohnnyDoorn                     14           309             18            0.06
Jonathon Love                  625        115492          45896           28.74
Joris Goosen                   478         96024          92319           33.54
Koen Derks                       5           125             82            0.04
Lars Willighagen                 1            15             10            0.00
Patrick Knight                  44         15901           5808            3.87
Raoul Grasman                    5           344              1            0.06
Ravi Selker                     41           456            236            0.12
Sacha Epskamp                    5           438            130            0.10
Simon Kucharsky                  1             1              1            0.00
Tahiraj                          6            17             13            0.01
Tim de Jong                     35          1185            828            0.36
TimKDJ                          43           629            346            0.17
akash raj                       22          8445           7891            2.91
akashraj                        55          3227           1139            0.78
akashrajkn                      42         13528           2500            2.85
alkight                          6           472            314            0.14
boutinb                          5            37             25            0.01
bruno boutin                   340         47826          36731           15.06
dropmann                       141         16659           4580            3.78
etothej                         13           102             14            0.02
jasp                             3          1920             24            0.35
mmarsman1                        1             4              4            0.00
neekro                           1             3              2            0.00
quentingronau                    6            25              6            0.01
vandenman                       18          1506            717            0.40
vankesteren                     14           213             51            0.05
¨Frans                           1             5              5            0.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Alexander Ly                  2          100.0         17.2                0.00
AlexanderLyNL                 6            5.3         16.7                0.00
Bruno Boutin                 61           30.0         11.3                0.00
Erik-Jan van Kestere        491           25.7         13.0               14.46
FransMeerhoff              6228           32.3         24.7               13.62
FransTestAccount              4            1.2         49.3                0.00
Janto91                       1          100.0         24.5                0.00
JohnnyDoorn                   4            1.3         19.6                0.00
Jonathon Love             41943           36.3         14.9               14.63
Joris Goosen              36117           37.6         11.5                2.92
Koen Derks                  119           95.2         13.9                2.52
Lars Willighagen             13           86.7          5.0                0.00
Patrick Knight             2066           13.0         38.6               38.92
Ravi Selker                   2            0.4         53.9                0.00
Sacha Epskamp                 4            0.9         65.9                0.00
Tahiraj                       5           29.4         45.8                0.00
Tim de Jong                 808           68.2          8.8                2.35
akash raj                  1321           15.6         19.3               13.10
akashrajkn                12166           89.9         11.4                9.28
bruno boutin              15851           33.1         14.7               11.75
dropmann                   9995           60.0         55.0                8.84
jasp                       1895           98.7         26.0               25.70
quentingronau                 2            8.0         60.5                0.00
vandenman                   250           16.6         10.7                2.40
vankesteren                 212           99.5         29.0               17.92
¨Frans                        5          100.0         45.8                0.00
```

**Comment by danielskatz** [comment]
@whedon generate pdf from branch JfA-JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JfA-JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02152/joss.02152/10.21105.joss.02152.pdf)

**Comment by danielskatz** [comment]
@koenderks - the use of "conform" in the paper seems incorrect.  Perhaps you can rephrase this sentence?

**Comment by danielskatz** [comment]
@whedon invite @drvinceknight as editor

**Comment by whedon** [comment]
@drvinceknight has been invited to edit this submission.

**Comment by arfon** [comment]
:wave: @koenderks - could you summarize here for the editor (and future reviewers) the specific aspects of the software that you would like reviewed here? (@koenderks and I have exchanged emails on this topic).

**Comment by koenderks** [comment]
Dear Arfon, dear (future) reviewer,

Thank you for taking the time to pre-review our submission. Hereby we would like to clarify which files build up JASP for Audit (JfA). JfA consists of three different types of files: QML files for the user-interface, R files to run the computations and display the results, and .jasp files that can be loaded from JASP’s internal data library that contain example analyses using the module. 

The files that are part of our submission to JOSS can be found below:

_QML files for the interface:_

Resources -> Audit -> qml -> auditBayesianWorkflow.qml
Resources -> Audit -> qml -> auditClassicalWorkflow.qml
Resources -> Audit -> qml -> auditBayesianPlanning.qml
Resources -> Audit -> qml -> auditClassicalPlanning.qml

_R files:_

JASP-Engine -> JASP -> R -> auditBayesianWorkflow.R
JASP-Engine -> JASP -> R -> auditClassicalWorkflow.R
JASP-Engine -> JASP -> R -> auditBayesianPlanning.R
JASP-Engine -> JASP -> R -> auditClassicalPlanning.R
JASP-Engine -> JASP -> R -> auditCommonFunctions.R
JASP-Engine -> JASP -> R -> auditCommonFunctionBayesian.R

_.jasp files:_

Resources -> Data sets -> Data library -> 10. Audit -> BuiltIt Monetary.jasp
Resources -> Data sets -> Data Library -> 10. Audit -> BuildIt Binary.jasp

Furthermore, as suggested by @danielskatz, I will rephrase the sentence containing the incorrect use of “conform” in the paper.

**Comment by drvinceknight** [comment]
@whedon set @drvinceknight as editor

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by drvinceknight** [comment]
@whedon assign @drvinceknight as editor

**Comment by whedon** [comment]
OK, the editor is @drvinceknight

**Comment by arfon** [comment]
Dear authors and reviewers

We wanted to notify you that in light of the current COVID-19 pandemic, JOSS has decided to suspend submission of new manuscripts and to handle existing manuscripts (such as this one) on a "best efforts basis". We understand that you may need to attend to more pressing issues than completing a review or updating a repository in response to a review. If this is the case, a quick note indicating that you need to put a "pause" on your involvement with a review would be appreciated but is not required.

Thanks in advance for your understanding.

_Arfon Smith, Editor in Chief, on behalf of the JOSS editorial team._

**Comment by arfon** [comment]
:wave: @drvinceknight - just a friendly reminder that we're waiting on you to identify some reviewers for this submission.

**Comment by koenderks** [comment]
Dear editors,

I hope you are well. I have rebased the code in the `JfA-JOSS` branch to be consistent with the latest version of JASP (version 0.12.2), which came out today. This version is currently available to download from `https://jasp-stats.org/download/` and contains the `Audit` module as described in our submission. The rebase does not influence the files for our submission.

Sincerely,
Koen Derks

**Comment by arfon** [comment]
Thanks @koenderks! Could you take a look at [this list of people](https://bit.ly/joss-reviewers) and suggest some potential reviewers?

**Comment by drvinceknight** [comment]
Sorry @arfon and @koenderks I'm still struggling with child care / work but I'll hopefully have more time from next week.

**Comment by drvinceknight** [comment]
@whedon generate pdf from branch JfA-JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JfA-JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02152/joss.02152/10.21105.joss.02152.pdf)

**Comment by drvinceknight** [comment]
@tylerwmarrs @davismcc @AvianaGlobal would you be able to review this paper for JOSS? It presents a library to carry out audits, of use both to researchers and practitioners.

**Comment by tylerwmarrs** [comment]
@drvinceknight Sorry, I cannot commit to reviewing this work.

**Comment by drvinceknight** [comment]
I understand, thanks for getting back to me. :+1:

**Comment by davismcc** [comment]
@drvinceknight This looks like an interesting paper, but I feel much better suited to reviewing papers in the bioinformatics/genomics area than for auditing like this paper, so I won't review this one. Next time!

**Comment by drvinceknight** [comment]
Thanks for letting me know @davismcc I understand. :)

**Comment by drvinceknight** [comment]
@koenderks I'm struggling to find reviewers for this, I have made some offline queries as well as the ones you see here. I'll continue to do that but if you had any suggestions for reviewers (including from this [list of potential reviewers](https://docs.google.com/spreadsheets/d/1PAPRJ63yq9aPC1COLjaQp8mHmEq3rZUzwUYxTulyu78/edit?usp=sharing)) please let me know.

**Comment by koenderks** [comment]
Dear @drvinceknight,

I hope you are well. I have rebased the code in the `JfA-JOSS` branch to be consistent with the latest version of JASP (version 0.13), which came out last Friday. 

In addition, I am happy to be able to tell you that we have found a potential reviewer for our submission. Dr. Trevor Stewart specializes in global audit policy, methodology, and technology, and is willing to act as a reviewer. Could you provide some advice on the correct procedure for including Dr. Stewart in the review process?

Sincerely,
Koen

**Comment by drvinceknight** [comment]
That's great, thanks @koenderks, does Dr Trevor Stewart have a GitHub account? If not let me know what their email address is (link me to their academic site perhaps) and I'll get in touch with them.

**Comment by koenderks** [comment]
Hi @drvinceknight, 

Regarding the GitHub account, I believe not. However, I can link you to the [academic site](http://raw.rutgers.edu/carlab/stewart.html). If I can be of any further help, please let me know.

**Comment by drvinceknight** [comment]
Thanks I'll drop them an email.

**Comment by trstewart212** [comment]
I am prepared to review, though it will be more for the auditing/statistical content than the code.

**Comment by drvinceknight** [comment]
That's fantastic, thanks @trstewart212.

**Comment by drvinceknight** [comment]
@whedon assign @trstewart212  as reviewer

**Comment by whedon** [comment]
OK, @trstewart212  is now a reviewer

**Comment by drvinceknight** [comment]
@trstewart212 once a second reviewer has been identified I will start the review.

**Comment by koenderks** [comment]
Dear @drvinceknight,

I am happy to let you know that we have found another reviewer for JASP for Audit. Ir. Ferry Geertman RE CISA (@fgeertman) is willing to act as a reviewer for this submission. 

Sincerely,
Koen

**Comment by trstewart212** [comment]
Please let me know what I am expected to do...

Trevor Stewart
175 West 13th Street, Apt 19D
New York, NY 10011
+1(212)929-0321
+1(917)402-9800 iPhone
trsny@verizon.net






On Sep 23, 2020, at 03:57, Koen Derks <notifications@github.com> wrote:


Dear @drvinceknight <https://github.com/drvinceknight>,

I am happy to let you know that we have found another reviewer for JASP for Audit. Ir. Ferry Geertman RE CISA (@fgeertman <https://github.com/fgeertman>) is willing to act as a reviewer for this submission.

Sincerely,
Koen

**Comment by drvinceknight** [comment]
@fgeertman can I confirm that you're happy to act as a reviewer for this paper?

**Comment by drvinceknight** [comment]
> Please let me know what I am expected to do...

Hi @trstewart212, we are close to starting now I hope and once that's the case I will let you know.

Thanks again to all for your patience.

**Comment by fgeertman** [comment]
Hi,

Yes, you can confirm

Kind Regards,

Ferry

Ferry Geertman
Managing Director - KEY Group
Mobile:  +31 6 8379 0414
Email: ferry.geertman@key-group.nl<mailto:ferry.geertman@key-group.nl>
[cid:image013.png@01D0836B.EB3E4390]

**Comment by drvinceknight** [comment]
@whedon add @fgeertman as reviewer

**Comment by whedon** [comment]
OK, @fgeertman is now a reviewer

**Comment by drvinceknight** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/2733.

**Comment by drvinceknight** [comment]
@fgeertman and @trstewart212 the review will take place on #2733. There are instructions over there as to what to do but if I can assist at all with any questions please let me know.
